### ISSUE TITLE
Free dockerhub organization are being phased out

### ISSUE THREAD

**Issue by wholtz** [mamba-org/micromamba-docker#276]
I received the following email from Docker:

> Docker is sunsetting Free Team organizations
> 
> Free Team organizations are a legacy subscription tier that no longer exists. This tier included many of the same features, rates, and functionality as a paid Docker Team subscription.
> 
> After reviewing the list of accounts that are members of legacy Free Team organizations, we’ve identified yours as potentially being one of them.
> 
> If you own a legacy Free Team organization, access to paid features — including private repositories — will be suspended on April 14, 2023 (11:59 pm UTC). Upgrade your subscription before April 14, 2023 to continue accessing your organization.
> 
> If you don’t upgrade to a paid subscription, Docker will retain your organization data for 30 days, after which it will be subject to deletion. During that time, you will maintain access to any images in your public repositories, though rate limitations will apply. At any point during the 30-day period, you can restore access to your organization account if you upgrade to a paid subscription. Visit our [FAQ](https://em.docker.com/NzkwLVNTQi0zNzUAAAGKf9auOxlLhT3vFReTAyFRlGHALJqz8cCKWRw2VZZGbkn9CmJYe3DcAykkkOatjvLH8S2TQ2OQfdxTScw=) for more information.

Without any action, this will break people's CI pipelines, as the old images will no longer be available on Dockerhub (at least not at the same location). Cost to get a paid team is $300/year.

There is still a free Team organization option if you are accepted into Docker's open source program. I applied to their open source program months ago and never heard back. I just noticed I had missed an email from Docker, saying that the process had changed and I need to reapply. So I have done that. Docker indicates that they try to process these applications within 30 days, but they are currently experience a high volume of applications.

I think we need to put some pressure on Docker delay this long enough for them to clear their open source program application queue. I can try to contact them and explain the situation, but I'm skeptical that I'll reach anyone with power to change this. We could really use some public pressure. Maybe I'll make a twitter account to work on this, but it would be great if someone more established on twitter (@wolfv  ?) could shine some light on this.

**Comment by maresb** [comment]
It doesn't solve the issue of historical images, but maybe we could switch over to GitHub Container Registry?

We should also be able to download the images from DockerHub and reupload them to GHCR. (I believe GHCR supports direct uploads.) Then it's just a matter of changing the domain.

**Comment by wholtz** [comment]
Yes, we should move to having multiple image repositories. That should be not too difficult and I should start on that.

But what I'm really worried about is breaking CI pipelines that pull our image from dockerhub.....

**Comment by wholtz** [comment]
https://github.com/docker/hub-feedback/issues/2314

**Comment by wholtz** [comment]
On the hub-feedback thread, a Docker employee has indicated nothing will be deleted while our application to the open source program is being processed. And if our application is denied, we will have 30 days from when we receive the rejection before deleting occurs.

**Comment by wholtz** [comment]
okay, that was fast...

> Congratulations! Your project has been approved for the Docker-Sponsored Open Source program. A Docker Team subscription will be allocated to the project organization Docker ID specified in your application within the next 3 weeks. In addition to a Team subscription, you now have:
> 
>     free autobuilds
> 
>     rate-limit removal for all users pulling public images from your project namespace
> 
>     special badging on Docker Hub (this will be visible within two weeks)
> 
> If you haven’t already, please take the time to update your project’s Hub pages to include a detailed project description, links to your project source code, as well as contributing guidelines, and a link to your organization’s website. Projects lacking this information may not receive the Docker Sponsored Open Source badging for their images on Docker Hub.
> 
> Membership in the Docker-Sponsored Open Source (DSOS) program is valid for one year. You will receive a reminder email from us when it is time to renew your membership.
> 
> We are thrilled to have you as an open-source partner. If you have any questions regarding the open source program please contact [opensource@docker.com](mailto:opensource@docker.com) and include the name of your project in the subject line. If you have questions regarding technical support please contact [support@docker.com](mailto:support@docker.com).
> 
> Thank you!

**Comment by wholtz** [comment]
And we are now pushing images to ghcr.io as well (#277). Only v1.3.1 is there (and future versions), since we are not loosing Dockerhub, I don't see a strong enough reason to copy older images over to ghcr.io.

**Comment by Vad1mo** [comment]
If, for some reason, the Docker-Sponsored Open-Source Program isn't working out, we are happy to help you out. We operate a Harbor-based [container registry as a service ](https://container-registry.com/)that has many benefits over most of the other registries out there.

There are also features regarding containerized image distribution, that might be valuable too as well.
